File: src/main/java/com/worldmanager/platform/kpi/data/models/kpi/Template.java
package com.worldmanager.platform.kpi.data.models.kpi;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

//@Entity
//@Table(name = "kpiTemplate")
public class Template {

//    @Id
//    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private String name;
    private String description;
    private Boolean isActive = false;

    private List<Section> sections = new ArrayList<>();

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String toString() {
        return String.format("KRI Template[id = %d, name='%s']", this.getId(), this.getName());
    }

    public Boolean getActive() {
        return isActive;
    }

    public void setActive(Boolean active) {
        isActive = active;
    }

    public List<Section> getSections() {
        return sections;
    }

    public void setSections(List<Section> sections) {
        this.sections = sections;
    }
}

File: src/main/scala/com/worldmanager/platform/kpi/scala/data/kpi/models/Template.scala
package com.worldmanager.platform.kpi.scala.data.kpi.models

import scala.beans.BeanProperty
import javax.validation.constraints.Size

class Template {
  @BeanProperty var id: Long = _
  @Size(min=1, max=256)
  @BeanProperty var name: String = _
  @BeanProperty var description: String = _
  @BeanProperty var isActive: Boolean = _
  @BeanProperty var sections: List[Section] = _

  override def toString = {
    "Template [id: %d, name = %s, isActive = %s]".format(id, name, isActive.toString)
  }
}
